report/schemas.api.md

```diff
@@ -399,6 +447,49 @@ export type ValidWorldRange = {
     yMax: number;
 };
 
+// @alpha (undocumented)
+export type Wearable = {
+    id: string;
+    description: I18N[];

```

Comment:
using arrays for this case is valid tho, we can add code validations to ensure uniqueness of keys, but keeping the schema portable. there are no "maps" in json schema, and records are not friendly to use as arrays are

Reply:
Can those validations be written in the JSON schema?

Reply:
I don't think there is a way to validate that in the JSON schema (please correct me if I'm wrong).

I personally think we should go with records to validate it correctly, or go with arrays and accept that it could happen that someone uses the same locale more than once.

I wouldn't add an extra layer of validations on the catalyst just for that.

Reply:
It is hard to me to articulate how bad have been records in the past. I tried to write about that here https://menduz.com/posts/2019.05.07 in the title `Avoid “maps” when possible.`. But I will leave this conversation to you to decide whether or not it makes sense in this case or not.

@nchamo the extra layer should exist in this repo, not in catalysts. Specifically, as a high order function or wrapper around

```ts
export const validate: ValidateFunction<Wearable> = generateValidator(schema)
```
would become
```ts
const schemaValidator: ValidateFunction<Wearable> = generateValidator(schema)
export const validate: ValidateFunction<Wearable> = (value: Wearable) => {
  schemaValidator(value) // i think it throws
  validateDuplicatedKeys(value.i18n) // throws also
}
```

Reply:
As long as the validations live here in this same repository it's okay for me, the only concern I have is leaving the schema so flexible that an entity can have extra fields with any value on them.

Reply:
Oh, great then! 👍